coderd/presets_test.go

```diff
@@ -17,58 +17,128 @@ import (
 func TestTemplateVersionPresets(t *testing.T) {
 	t.Parallel()
 
-	givenPreset := codersdk.Preset{
-		Name: "My Preset",
-		Parameters: []codersdk.PresetParameter{
-			{
-				Name:  "preset_param1",
-				Value: "A1B2C3",
+	testCases := []struct {
+		name    string
+		presets []codersdk.Preset
+	}{
+		{
+			name:    "no presets",
+			presets: []codersdk.Preset{},
+		},
+		{
+			name: "single preset with parameters",
+			presets: []codersdk.Preset{
+				{
+					Name: "My Preset",
+					Parameters: []codersdk.PresetParameter{
+						{
+							Name:  "preset_param1",
+							Value: "A1B2C3",
+						},
+						{
+							Name:  "preset_param2",
+							Value: "D4E5F6",
+						},
+					},
+				},
 			},
-			{
-				Name:  "preset_param2",
-				Value: "D4E5F6",
+		},
+		{
+			name: "multiple presets with overlapping parameters",
```

Comment:
Might be useful to have a test which has only a partial overlap in parameters, too.

Reply:
We do, on line 52:

```
						{
							Name:  "shared_param",
							Value: "value1",
						},
```

Reply:
Whoops 🤦